SecurityException in Fragment_AlarmDetails_Main

### Crash report received from Google Play

**Device information:**
 - Device model: Samsung Galaxy A01
 - Android version: Android 10 (SDK 29)
 - App version: v1.3.3
 - Downloaded from (GitHub/Google Play): Google Play

**Stack trace:**

```java
java.lang.SecurityException: 
  at android.os.Parcel.createException (Parcel.java:2088)
  at android.os.Parcel.readException (Parcel.java:2056)
  at android.os.Parcel.readException (Parcel.java:2004)
  at android.app.IActivityManager$Stub$Proxy.getContentProvider (IActivityManager.java:5759)
  at android.app.ActivityThread.acquireProvider (ActivityThread.java:7035)
  at android.app.ContextImpl$ApplicationContentResolver.acquireUnstableProvider (ContextImpl.java:2893)
  at android.content.ContentResolver.acquireUnstableProvider (ContentResolver.java:2124)
  at android.content.ContentResolver.query (ContentResolver.java:935)
  at android.content.ContentResolver.query (ContentResolver.java:887)
  at android.content.ContentResolver.query (ContentResolver.java:843)
  at in.basulabs.shakealarmclock.Fragment_AlarmDetails_Main.displayAlarmTone (Fragment_AlarmDetails_Main.java:329)
  at in.basulabs.shakealarmclock.Fragment_AlarmDetails_Main.onCreateView (Fragment_AlarmDetails_Main.java:141)
  at androidx.fragment.app.Fragment.performCreateView (Fragment.java:2698)
  at androidx.fragment.app.FragmentStateManager.createView (FragmentStateManager.java:310)
  at androidx.fragment.app.FragmentManager.moveToState (FragmentManager.java:1185)
  at androidx.fragment.app.FragmentManager.moveToState (FragmentManager.java:1354)
  at androidx.fragment.app.FragmentManager.moveFragmentToExpectedState (FragmentManager.java:1432)
  at androidx.fragment.app.FragmentManager.moveToState (FragmentManager.java:1495)
  at androidx.fragment.app.FragmentManager.dispatchStateChange (FragmentManager.java:2617)
  at androidx.fragment.app.FragmentManager.dispatchActivityCreated (FragmentManager.java:2569)
  at androidx.fragment.app.FragmentController.dispatchActivityCreated (FragmentController.java:247)
  at androidx.fragment.app.FragmentActivity.onStart (FragmentActivity.java:541)
  at androidx.appcompat.app.AppCompatActivity.onStart (AppCompatActivity.java:210)
  at android.app.Instrumentation.callActivityOnStart (Instrumentation.java:1433)
  at android.app.Activity.performStart (Activity.java:7978)
  at android.app.ActivityThread.handleStartActivity (ActivityThread.java:3512)
  at android.app.servertransaction.TransactionExecutor.performLifecycleSequence (TransactionExecutor.java:221)
  at android.app.servertransaction.TransactionExecutor.cycleToPath (TransactionExecutor.java:201)
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:173)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:97)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2175)
  at android.os.Handler.dispatchMessage (Handler.java:107)
  at android.os.Looper.loop (Looper.java:237)
  at android.app.ActivityThread.main (ActivityThread.java:7860)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:493)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1068)
Caused by: android.os.RemoteException: 
  at com.android.server.am.ActivityManagerService.getContentProviderImpl (ActivityManagerService.java:8218)
  at com.android.server.am.ActivityManagerService.getContentProviderImpl (ActivityManagerService.java:8064)
  at com.android.server.am.ActivityManagerService.getContentProvider (ActivityManagerService.java:8694)
  at android.app.IActivityManager$Stub.onTransact (IActivityManager.java:2409)
  at com.android.server.am.ActivityManagerService.onTransact (ActivityManagerService.java:3324)
```

**Possible explanation:**
The exception is raised at this line:
https://github.com/WrichikBasu/ShakeAlarmClock/blob/d013432b14b8642fa4dc802d9dd6d615bad54beb/app/src/main/java/in/basulabs/shakealarmclock/Fragment_AlarmDetails_Main.java#L329

There is some problem with the read external storage permission.

--- a/app/src/main/java/in/basulabs/shakealarmclock/Fragment_AlarmDetails_Main.java
+++ b/app/src/main/java/in/basulabs/shakealarmclock/Fragment_AlarmDetails_Main.java
@@ -316,7 +316,7 @@ public class Fragment_AlarmDetails_Main extends Fragment implements View.OnClick
 	//----------------------------------------------------------------------------------------------------
 
 	/**
-	 * Displays the alarm tone in {@link #alarmToneTV}.
+	 * Displays the alarm tone file name in {@link #alarmToneTV}.
 	 */
 	private void displayAlarmTone() {
 
@@ -327,10 +327,10 @@ public class Fragment_AlarmDetails_Main extends Fragment implements View.OnClick
 
 			String fileName = null;
 
-			try (Cursor cursor = requireContext().getContentResolver()
-					.query(viewModel.getAlarmToneUri(), null, null, null, null)) {
+			try {
+				try (Cursor cursor = requireContext().getContentResolver()
+						.query(viewModel.getAlarmToneUri(), null, null, null, null)) {
 
-				try {
 					if (cursor != null && cursor.getCount() > 0 && cursor.moveToFirst()) {
 
 						int index = cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME);
@@ -339,9 +339,17 @@ public class Fragment_AlarmDetails_Main extends Fragment implements View.OnClick
 						} else {
 							fileName = cursor.getString(RingtoneManager.TITLE_COLUMN_INDEX);
 						}
+					} else {
+						viewModel.setAlarmToneUri(Settings.System.DEFAULT_ALARM_ALERT_URI);
+						alarmToneTV.setText(R.string.defaultAlarmToneText);
+						return;
 					}
-				} catch (Exception ignored) {
 				}
+			} catch (java.lang.SecurityException se){
+				viewModel.setAlarmToneUri(Settings.System.DEFAULT_ALARM_ALERT_URI);
+				alarmToneTV.setText(R.string.defaultAlarmToneText);
+				return;
+			} catch (Exception ignored) {
 			}
 
 			if (fileName != null) {

